src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -330,6 +417,258 @@ public void jobFinished () {
         NotifyUsersForSubscriptionJob.createNotification("deployment-updated", deployment.id, message);
     }
 
+
+    private void replaceEC2Servers() {
+        try {
+            // Track any previous instances running for the server we're deploying to in order to de-register and
+            // terminate them later.
+            List<EC2InstanceSummary> previousInstances = otpServer.retrieveEC2InstanceSummaries();
+            // First start graph-building instance and wait for graph to successfully build.
+            status.message = "Starting up graph building EC2 instance";
+            List<Instance> instances = startEC2Instances(1);
+            status.message = "Waiting for graph build to complete...";
+            MonitorServerStatusJob monitorInitialServerJob = new MonitorServerStatusJob(owner, deployment, instances.get(0), otpServer);
+            monitorInitialServerJob.run();
+            status.update("Graph build is complete!", 50);
+            // Spin up remaining servers which will download the graph from S3.
+            int remainingServerCount = otpServer.ec2Info.instanceCount <= 0 ? 0 : otpServer.ec2Info.instanceCount - 1;
+            if (remainingServerCount > 0) {
+                // Spin up remaining EC2 instances.
+                status.message = String.format("Spinning up remaining %d instance(s).", remainingServerCount);
+                List<Instance> remainingInstances = startEC2Instances(remainingServerCount);
+                instances.addAll(remainingInstances);
+                // Create new thread pool to monitor server setup so that the servers are monitored in parallel.
+                ExecutorService service = Executors.newFixedThreadPool(remainingServerCount);
+                for (Instance instance : remainingInstances) {
+                    // Note: new instances are added
+                    MonitorServerStatusJob monitorServerStatusJob = new MonitorServerStatusJob(owner, deployment, instance, otpServer);
+                    service.submit(monitorServerStatusJob);
+                }
+                // Shutdown thread pool once the jobs are completed and wait for its termination. Once terminated, we can
+                // consider the servers up and running (or they have failed to initialize properly).
+                service.shutdown();
+                service.awaitTermination(4, TimeUnit.HOURS);
+            }
+            String finalMessage = "Server setup is complete!";
+            // Get EC2 servers running that are associated with this server.
+            List<String> instanceIds = previousInstances.stream()
```

Comment:
I think it'd be good to have a variable name that continues to indicate that these are previous instances. Maybe `previousInstanceIds`.